src/NuGet.Core/NuGet.Common/Telemetry/Performance/ActionEventBase.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace NuGet.Common
+{
+    /// <summary>
+    /// Base class to generate telemetry data for nuget operations like install, udpate or restore.
+    /// </summary>
+    public abstract class ActionEventBase
+    {
+        public ActionEventBase(
+            string operationId,
+            string[] projectIds,
+            DateTime startTime,
+            NugetOperationStatus status,
+            string statusMessage,
+            int packageCount,
+            DateTime endTime,
+            double duration)
+        {
+            OperationId = operationId;
+            ProjectIds = projectIds;
+            PackagesCount = packageCount;
+            Status = status;
+            StatusMessage = statusMessage;
+            StartTime = startTime;
+            EndTime = endTime;
+            Duration = duration;
+        }
+
+        public string OperationId { get; set; }
```

Comment:
No reason, earlier i had a different design where i needed those setter but then i changed it. But forgot to remove these setter. But I'll remove these.